### ISSUE TITLE
Release as a capacitor plugin

### ISSUE THREAD

**Issue by ptmkenny** [j3k0/cordova-plugin-purchase#1328]
I know the name of this plugin is **cordova**-plugin-purchase, but is there any chance of a native capacitor plugin being released?

**Comment by HarelM** [comment]
This plugin can be used in capacitor...

**Comment by ptmkenny** [comment]
@HarelM Yes, capacitor supports cordova plugins. However, this is a feature request to release a native capacitor plugin instead of using the cordova compatibility layer.

**Comment by HarelM** [comment]
Since capacitor supports cordova I don't see a lot of benefit opening an issue here, maintaining a capacitor plugin is a lot of work...
What's the added value for the maintainer here to do it? (I'm not maintain this package BTW)

**Comment by ptmkenny** [comment]
The benefit is that it will bring more users to the plugin.  Capacitor passed Cordova in npm downloads in 2021 and  cordova is clearly in decline, while usage of Capacitor is increasing.  This is partly because Capacitor has a more modern architecture and partly because Ionic is taking better care of Capacitor (it's part of their main business) than Adobe has with Cordova.

https://npmtrends.com/@capacitor/core-vs-cordova

Show downloads over the past 5 years and the trend is quite clear.

**Comment by HarelM** [comment]
All that is true, but it since this plugin works with capacitor and there's no alternative in capacitor I don't see any added value. Right now this plugin supports both capacitor and cordova. What's the added value of limiting it to only capacitor?

**Comment by ptmkenny** [comment]
This is a feature request to add a capacitor version of the plugin.  I didn't say anything about getting rid of the cordova version.

Because there is no capacitor version, users of recent versions of Ionic who are using Capacitor need to install the wrapper through [awesome-cordova-plugins](https://github.com/danielsogl/awesome-cordova-plugins) and then use Capacitor's cordova support.  Yes, this works, but it's using a wrapper to run legacy code.  So it would be better to have a Capacitor plugin.

I hope a maintainer can comment on plans for direct Capacitor support.

For example, is there a plan to add a support, or if there are no plans, would an MR that adds support be considered?

**Comment by j3k0** [comment]
@ptmkenny We don't have internal experience with Capacitor, but checking what would be required is in our plans. I believe most of the code can be shared, as the layer of communication from JS to Native (and back) is quite minimal in this plugin.

**Comment by j3k0** [comment]
> MR that adds support be considered?

Obviously yes, however I would highly recommend to wait for #1330 - as I'm rewriting large chunk of the code to support Google Billing Library v5, StoreKit 2, and other payment processors.

**Comment by neonish** [comment]
@ptmkenny @j3k0 any ETA for this?

v13 breaks compatibility with [@awesome-cordova-plugins/in-app-purchase-2](https://www.npmjs.com/package/@awesome-cordova-plugins/in-app-purchase-2) which is needed to use this plugin with capacitor. See [here](https://github.com/danielsogl/awesome-cordova-plugins/issues/4457).

**Comment by ptmkenny** [comment]
@neonish I have not started on this and I am not sure when I will have time to start.

**Comment by awltr** [comment]
Also had issues with v13 and needed to downgrade. In addition, the setup is quite confusing. I'm still not sure if mine is correct but had to add @ionic-native/in-app-purchase-2 and @ionic-native/core because it's a peer dependency. So 300kb additional bundle size without even having an ionic project and being sure what of it is needed and what is not.
IAP is a key feature of app development and as already mentioned here, capacitor is the disruptive technology to cordova. So having this as native plugin would be the right way imho.
